src/coreclr/jit/codegenxarch.cpp

```diff
@@ -504,7 +504,22 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
 
             if (vecCon->IsZero())
             {
-                if ((attr != EA_32BYTE) || compiler->compOpportunisticallyDependsOn(InstructionSet_AVX))
+                bool isSupported;
+
+                if (attr == EA_32BYTE)
+                {
+                    isSupported = compiler->compOpportunisticallyDependsOn(InstructionSet_AVX);
+                }
+                else if (attr == EA_64BYTE)
+                {
+                    isSupported = compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F);
+                }
+                else
+                {
+                    assert((attr == EA_8BYTE) || (attr == EA_16BYTE));
+                    isSupported = true;
+                }
```

Comment:
It's probably worth making this a `switch(attr)` or otherwise prioritizing `if (attr <= EA_16BYTE)` first.

That will help throughput on the crossgen case and allow the compiler to better optimize for the expected inputs.